### PR TITLE
[3203] - Add Provider search to Organisation Index page

### PR DESCRIPTION
**Pull request by aliuk2012**
### Context
Both org#index pages have a big hit on our API. We should had search then pagination to improve the performance (and UX)

### Changes proposed in this pull request
- Add ability to search by provider name or provider code. Includes autocomplete functionality. This is available to admin users only

provider_search

### Checklist
- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally
- [ ] Product Review

### CODE REVIEW

File: spec/features/providers/search_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  let(:current_recruitment_cycle) { build(:recruitment_cycle) }
+  let(:rollover) { false }
+
+  before do
+    stub_omniauth
+    stub_api_v2_request(
+      "/recruitment_cycles/#{current_recruitment_cycle.year}",
+      current_recruitment_cycle.to_jsonapi,
+    )
+
```
Comment (dankmitchell) [inline spec/features/providers/search_spec.rb:13]: Extra white space...

File: app/controllers/providers_controller.rb [hunk 2]
```diff
@@ -7,9 +7,8 @@ class ProvidersController < ApplicationController
   before_action :build_training_provider, only: %i[training_provider_courses]
 
   def index
-    @providers = Provider
-      .where(recruitment_cycle_year: Settings.current_cycle)
-      .all
+    @providers = Kaminari.paginate_array(Provider.where(recruitment_cycle_year: Settings.current_cycle).all)
+                          .page(params[:page] || 1).per(params[:per_page] || 25)
```
Comment (dankmitchell) [inline app/controllers/providers_controller.rb:11]: Is 25 too many? i.e. still too slow? Can we go with 10?

**Comment by TobyRet** [comment]
[Add search to org#index](https://trello.com/c/xS5t1W0N/3203-add-search-to-orgindex)

File: app/webpacker/scripts/__snapshots__/autocomplete.spec.js.snap [hunk 3]
```diff
@@ -7,52 +7,14 @@ exports[`Autocomplete initAutocomplete should instantiate an autocomplete 1`] =
...
-      <ul
-        class="autocomplete__menu autocomplete__menu--inline autocomplete__menu--hidden"
-        id="input__listbox"
-        role="listbox"
-      >
-        
-      </ul>
-    </div>
-  </div>
+  />
```
Comment (TobyRet) [inline app/webpacker/scripts/__snapshots__/autocomplete.spec.js.snap:17]: Looking into this ...

Comment (TobyRet) [reply app/webpacker/scripts/__snapshots__/autocomplete.spec.js.snap]: Fixed now. There was an import error.

**Pull request merged**